### PR TITLE
Improve preprocessing of field and option labels

### LINKED ISSUE #70: [FEATURE] Improve `field_label` metadata for checkbox fields
# Feature Request Description
Currently when checkbox fields are expanded in the metadata to have one row per checkbox option the field label from the checkbox field is carried over to all new rows. Field labels for each new row should be updated to include the option label in addition to the field label.

This addresses the conversation in https://github.com/CHOP-CGTDataOps/REDCapTidieR/pull/65#issuecomment-1306177174 

# Proposed Solution
Update `update_field_names()` to append field labels with option labels. See https://github.com/CHOP-CGTDataOps/REDCapTidieR/pull/65#issuecomment-1306177174 for example output.

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR updates the processing we apply to field labels and option labels for checkbox/radio/dropdown fields based on the decisions reached here https://github.com/CHOP-CGTDataOps/REDCapTidieR/pull/73#issuecomment-1314369762.

# Proposed Changes 
List changes below in bullet format:

- Alter `update_field_names()` to edit `db_metadata$field_label` in place by:
    - appending checkbox option labels after the field label
    - removing html tags and field embedding logic
- Update `parse_labels()` to remove html + field embedding logic from labels
    - This flows through to `redcap_metadata` as well as factor labels in `redcap_data`
- Update test cases for both `update_field_names()` and `parse_labels()`
- Update test data since this PR modifies functions called directly in `create_test_data.R`

### Issue Addressed
Closes #70

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

**Review by rsh52 — CHANGES_REQUESTED** [review]
One thing to change and just a couple comments otherwise. Looks good, getting closer to done!

**Review by skadauke — APPROVED** [review]
This looks great to me! Love the tests!

File: R/utils.R [hunk 1]
```diff
@@ -169,77 +172,100 @@ parse_labels <- function(string, return_vector = FALSE) {
...
+
+      # Build updated field names and labs
+      clean_names <- paste(
+        out$field_name[[i]],
+        tolower(parsed_labs$raw),
+        sep = "___"
+      )
+
+      # If field_label or options_labels are missing don't label
+      if (is.na(out$field_label[[i]]) || any(is.na(parsed_labs$label))) {
```
[the remark this answers is not in the document]
Comment (ezraporter) [reply R/utils.R:231]: I'll revert that. Not sure I actually meant to make a change here. Was just coding on the fly.

I usually put `[[` when I expect the result to be atomic as opposed to vector/list but that may not even be the right thing to do.

File: R/utils.R [hunk 2]
```diff
@@ -410,3 +436,25 @@ multi_choice_to_labels <- function(db_data, db_metadata) {
 get_project_id_field <- function(data) {
```
[the remark this answers is not in the document]
Comment (ezraporter) [reply R/utils.R:436]: Good call!

File: tests/testthat/test-utils.R [hunk 3]
```diff
@@ -66,7 +66,7 @@ test_that("multi_choice_to_labels works", {
 
 test_that("parse_labels works", {
 
-  valid_string <- "choice_1, one | choice_2, two | choice_3, three"
+  valid_string <- "choice_1, one | choice_2, two {abc} | choice_3, <b>three</b>"
```
[the remark this answers is not in the document]
Comment (ezraporter) [reply tests/testthat/test-utils.R:69]: Will do this as well!

**Review by rsh52 — APPROVED** [review]

**Comment by rsh52** [comment]
So long as the checks pass the changes all look good to me! 🤞 🎉
